### PR TITLE
KAFKA-6627: Prevent config default values overriding ones specified through --producer-property on command line.

### PR DESCRIPTION
**Pull request by likan999**
In ConsoleProducer, extraProducerProps (options specified in --producer-property) are applied first, then overriden (thus ignored) by the values returned by command line parser (default values are used for ones not specified from command line). This patch fixes it so that it doesn't override the existing value set by --producer-property if it is not explicitly specified.

The contribution is my original work and I license the work to the project under the project's open source license.
    
Signed-off-by: Kan Li <likan@uber.com>

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*
Testing strategy:
For producer, print out acks value in KafkaProducer.configureAcks, start producer using --producer-property aks=all, then observe the difference with and without the patch.
For consumer, print out isolationLevel in KafkaConsumer, start consumer using  --consumer-property isolation.level=read_committed, then observe the difference with and without the patch.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by harshach** [comment]
@likan999 looks good. Can you also check if we should apply these changes to console-consumer as well?

File: core/src/main/scala/kafka/utils/ConfigUtils.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+object ConfigUtils {
+  /**
+    * Merge the options into {@code props} for key {@code key}, with the following precedence, from high to low:
+    * 1) if {@code spec} is specified on {@code options} explicitly, use the value;
+    * 2) if {@code props} already has {@code key} set, keep it;
+    * 3) otherwise, use the default value of {@code spec}.
+    * A {@code null} value means to remove {@code key} from the {@code props}.
+    */
+  def maybeMergeOptions[V](props: Properties, key: String, options: OptionSet, spec: OptionSpec[V]) {
```
Comment (harshach) [inline core/src/main/scala/kafka/utils/ConfigUtils.scala]: can you move this method to CommandLineUtils

**Comment by likan999** [comment]
Moved the method into CommandLineUtils so that it doesn't introduce a new file.

Also tested it against consumer and without the diff --consumer-property isolation.level=read_committed doesn't work, and works with the diff.

Consumer is kind of different to producer as the command line flags don't have many overlap with the ones specified through --consumer-property. isolation.level is one of a few overlaps, and other overlaps (group_id, e.g.) are handled specifically.

**Comment by likan999** [comment]
@harshach could you TAL? Thanks.

**Review by harshach — APPROVED** [review]

**Comment by hachikuji** [comment]
@likan999 @harshach Note the needed fix above. It's important to check build failures. Thanks!

**Pull request merged**